src/org/traccar/protocol/Jt600ProtocolDecoder.java

```diff
@@ -256,11 +259,29 @@ private Position decodeU01(String sentence, Channel channel, SocketAddress remot
         position.set(Position.KEY_CID, parser.nextInt());
         position.set(Position.KEY_LAC, parser.nextInt());
         position.set(Position.KEY_GSM, parser.nextInt());
-        position.set(Position.KEY_ODOMETER, parser.nextLong() * 1000);
+        position.set(Position.KEY_ODOMETER, parser.nextLong() * 100);
         position.set(Position.KEY_INDEX, parser.nextInt());
 
-        return position;
+        switch (messageType) {
+            case "U01":
+            case "U02":
+            case "U03":
+                String checkSum = parser.next();
+                int calculatedCheckSum = checkSum(sentence.substring(1, sentence.length() - checkSum.length() - 1));
```

Comment:
I am just following protocol documentation. It clearly says that:

> When server center received the gps data (e.g. U01,U02,U03,U07),and check if the checksum
is correct or not. If it’s correct, so send (S39) command to tracker, and tracker send the next
GPS data; if incorrect, so tracker will resend the last GPS data again until the checksum is ok

Maybe there can be a case, when it does not match. I don't know.